pinot-common/src/main/java/org/apache/pinot/common/function/scalar/MapFunctions.java

```diff
@@ -0,0 +1,67 @@
+package org.apache.pinot.common.function.scalar;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.pinot.spi.annotations.ScalarFunction;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+public class MapFunctions {
+    private MapFunctions() {
+    }
+
+    /**
+     * Internal Helper Function to convert Json String Values to Java Maps
+     * @param jsonString
+     * @return converted Java Map from JSON String
+     */
+    public static Map stringToMap(String jsonString) {
```

Comment:
You should be able to use `JsonUtils.jsonNodeToMap()`

Reply:
Redid implementation with a LinkedHashMap, is this function still usable? I further noticed that jsonUtils' jsonNodeToMap takes in a JSON Object whereas this new implementation is string based, do you have any thoughts on to still use jsonNodeToMap() or maybe add in a function there that takes string-based input?

Reply:
Any specific reason why you need a `LinkedHashMap`?
`JsonUtils.stringToObject()` can be used in order to deserialize to a custom object